src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/category/CategoriesFieldset.java

```diff
@@ -53,39 +53,71 @@ public boolean canEdit() {
 
   @CheckForNull
   public CategoryPill findCategoryWith(final String categoryName) {
-    CategoryPill result = null;
+    WebElement category = getCategoryWith(categoryName);
+    return category == null ? null : new CategoryPill(category);
```

Comment:
Perhaps you can use `Nullifier` from `uk.kludje:kludje-core:0.8`:
`Nullifier.eval(category, CategoryPill:new);`